src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/GenotypingEngine.java

```diff
@@ -339,27 +325,18 @@ private OutputAlleleSubset(final int count, final Allele[] alleles, final int[]
         }
 
         private List<Allele> outputAlleles(final Allele referenceAllele) {
-            final ArrayList<Allele> result = new ArrayList<>(count + 1);
-            result.add(referenceAllele);
-            for (int i = 0; i < count; i++) {
-                result.add(alleles[i]);
-            }
-            return result;
+            return Stream.concat(Stream.of(referenceAllele), Arrays.stream(alleles, 0, count)).collect(Collectors.toList());
```

Comment:
I agree that one should be pragmatic.  For me personally the streaming one-liner is faster to parse in this case because there aren't extra lines for constructing and returning and because elements of the result are contiguous in the code.